src/client.cpp

```diff
@@ -194,22 +194,25 @@ bool hasIntMember(const rapidjson::Value& value, const char* name)
     return value.HasMember(name)  &&  value[name].IsInt();
 }
 
+#define CREATE_SESSION() \
+    CurlSessionPtr sessionPtr = createSession();\
+    if (!sessionPtr)\
+        return makeError();\
+    CurlSession& session = *sessionPtr;
```

Comment:
I do not like that variable that is used further is not transparent to the user. I think it can be re-done so that variable declaration is outside of the macro. Internally it can have another hidden variable that is "returned" by the macro if it is valid. I would not say what we had before was that bad. a bit more code, but nothing too wordy or complicated.